src/librustc_typeck/check/demand.rs

```diff
@@ -250,9 +250,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
     // This function checks if the method isn't static and takes other arguments than `self`.
```

Comment:
The doc comment suggests to me this function is misused, because it doesn't actually check that.
It doesn't check `fn_has_self_parameter` at all.

Reply:
It also use at https://github.com/rust-lang/rust-clippy/blob/c211cea3e99e04c2980a853b6637de22881b72eb/clippy_lints/src/len_zero.rs#L127.
I think if it does have problems, we should fix them in another PR.